Machine.Specifications.Should/ShouldExtensionMethods.cs

```diff
@@ -620,11 +620,11 @@ public static void ShouldBeLike(this object obj, object expected)
             }
         }
 
-        static IEnumerable<SpecificationException> ShouldBeLikeInternal(object obj, object expected, string nodeName, List<object> visited)
+        static IEnumerable<SpecificationException> ShouldBeLikeInternal(object obj, object expected, string nodeName, HashSet<object> visited)
         {
             if (IsReferenceTypeNotNullOrString(obj) && IsReferenceTypeNotNullOrString(expected))
             {
-                if (visited.Any(o => ReferenceEquals(o, expected)))
+                if (visited.Contains(expected))
```

Comment:
Oh. Seems to be fixed in following commit (`ReferenceEquals()` calls in `SimpleTuple`).